lib/google_api_client.rb

```diff
@@ -0,0 +1,49 @@
+require 'google/apis/drive_v3'
+require 'googleauth'
+require 'google/apis/analytics_v3'
+require 'google/apis/sheets_v4'
+
+class GoogleApiClient
+  def initialize(options = {auth_file: nil})
+    @auth_file = options[:auth_file]
+
+    raise "missing auth_file" if @auth_file.nil?
+  end
+
+  def drive_file
```

Comment:
I would think `drive_file` would be an instance, rather than the class itself.

Reply:
Yeah I had that inclination too...I was thinking about how it's used and felt like:

`metadata = @google_api_client.drive_file.new(foo: 'bar')` felt nicer than.

```ruby
metadata = @google_api_client.drive_file
metadata.foo = "bar"
metadata.baz = "bing"
```

Maybe exposing it as a class-constant or method is more palatable ?

```
metadata = GoogleApiClient::DriveFile.new(foo: 'bar')
#or 
metadata = GoogleApiClient.drive_file.new(foo: 'bar')
```
Those seem nice because, since there isn't any instance-state used...the class name space feels more appropriate. What do you think?

Reply:
Yeah, I like the class name space. And I kinda prefer the `::DriveFile` personally. YMMV!